api/src/main/java/org/openmrs/api/db/hibernate/ImmutableEntityInterceptor.java

```diff
@@ -114,4 +117,32 @@ public boolean onFlushDirty(Object entity, Serializable id, Object[] currentStat
 		
 		return false;
 	}
+
+	/**
+	 * @return true if the given property is allowed to be mutable
+	 */
+	private boolean isMutableProperty(String property) {
+		if (ArrayUtils.contains(getMutablePropertyNames(), property)) {
+			return true;
+		}
+		String[] threadMutable = additionalMutableProperties.get();
+		if (threadMutable != null && ArrayUtils.contains(threadMutable, property)) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * @param properties any additional properties that one wishes to make mutable for a given thread
+	 */
+	public void addMutablePropertiesForThread(String... properties) {
+		additionalMutableProperties.set(properties);
+	}
+
+	/**
+	 * If any additional properties were added for a given thread, this removes and cleans up
+	 */
+	public void removeMutablePropertyForThread() {
+		additionalMutableProperties.remove();
+	}
```

Comment:
```suggestion
	public void removeMutablePropertiesForThread() {
		additionalMutableProperties.remove();
	}
```

Reply:
Thanks @ibacher - I've made those suggested changes.  To be honest, this feels like a hack, and there is likely a way to avoid the need for this with some heavy lifting or more in-depth knowledge of hibernate and such in the sync module, but I wasn't able to easily do so without making what felt like much more disruptive and error prone changes to the sync module.  Given this is a fairly innocuous change that should have pretty much no impact on any existing code or normal usage, it seemed like a reasonable solution.